src/grblogtools/api.py

```diff
@@ -60,26 +65,34 @@ def progress(self, section="nodelog") -> dict:
         )
 
     def common_log_data(self):
-        """Extract summary data to be joined to progress logs."""
-
-        # The summary data is joined to progress log on LogFilePath + LogNumber.
-        # This method separately calls summary() whenever a progress log is created,
-        # so perhaps it should be cached. The cache would need to be invalidated
-        # if parse() is ever called again.
-        common_columns = [
-            "LogFilePath",
-            "LogNumber",
-            "Log",
-            "ModelFilePath",
-            "ModelFile",
-            "Model",
-            "Seed",
-            "Version",
-        ]
-        summary = self.summary()
-        return summary.loc[:, summary.columns.isin(common_columns)]
-
-    def summary(self):
+        """Extract common data to be joined to progress and summary dataframes.
+        This could be cached in future and invalidated by .parse()"""
```

Comment:
This is minor. But following Google Style, the doctoring should look like:
```suggestion
        """Extract common data to be joined to progress and summary dataframes.
        
        This could be cached in future and invalidated by .parse().
       """
```